docs/ui/Toc.tsx

```diff
@@ -0,0 +1,98 @@
+'use client';
+
+import { useEffect, useRef, useState } from 'react';
+import { createPortal } from 'react-dom';
+
+import { Link, List } from '@marigold/components';
+
+export interface TocProps {
+  items: string;
+  activeItem?: string;
+}
+
+export const Toc = ({ items }: TocProps) => {
+  const elements = JSON.parse(items) as {
+    anchor: string;
+    title: string;
+    id: string;
+  }[];
+
+  const [, setIsMounted] = useState(false);
+
+  const ref = useRef<Element>();
+
+  const itemId = elements.map(item => item.id);
+  const activeItem = useActiveItem(itemId);
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      ref.current = document.querySelector('#toc') || undefined;
+      setIsMounted(true);
+    }
+  }, []);
+
+  if (!ref.current || elements.length === 0) {
+    return null;
+  }
+
+  const TocPortal = () => (
+    <div className="fixed">
+      <List as="ul">
+        On This Page
+        {elements.map((i: { title: string; anchor: string; id: string }) => (
+          <List.Item key={i.title}>
+            <Link
+              variant="toc"
+              href={i.anchor}
+              key={i.id}
```

Comment:
Is this key needed? I have already a key from List.Item.

Reply:
Ah.. Überbleibsel from trying stuff out :D yes can removed